util/repos/repos.go

```diff
@@ -0,0 +1,125 @@
+package repos
+
+import (
+	"io/ioutil"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// Client is a generic git client interface
+type Client interface {
```

Comment:
I think it is better to add HelmRepoServerService to `argocd-repo-server` and implement helm specific apis instead of trying to encapsulate interaction with git and helm repo. Here are the use cases where it makes things unnecessarily complicated:

- `/api/v1/repositories/{repo}/apps` (ListApps) is not efficient. In case of helm `argocd-server` would query repo server 6 times with following filters: "*app.yaml", "*components/params.libsonnet", "*Chart.yaml", "kustomization.yaml", "kustomization.yml", "Kustomization". Instead `argocd-server` need an API which returns list of charts in given helm repository. 

- The ListDir gRPC is supposed to be executed for a specific revision, but in helm repo each chart has separate revision. 

- During helm chart validation `argocd-server` executes GetFile gRPC which downloads chart and check for presents of Chart.yaml, this is not necessary since it only need to make sure chart is mentioned in registry.